react-app/src/features/package/index.tsx

```diff
@@ -54,16 +51,32 @@ export const DetailsContent: FC<{ id: string }> = ({ id }) => {
   );
 };
 
-export const Details = () => {
-  const { id, authority } = useParams<{
-    id: string;
-    authority: Authority;
-  }>();
-
+export const packageDetailsLoader = async ({ params }: LoaderFunctionArgs) => {
+  const { id, authority } = params;
   if (id === undefined || authority === undefined) {
-    return <Navigate to="/dashboard" />;
+    return redirect("/dashboard");
   }
 
+  try {
+    const packageResult = await getItem(id);
+    if (!packageResult || packageResult?._source?.deleted === true) {
+      return redirect("/dashboard");
+    }
+  } catch (error) {
+    console.log("Error: ", error);
+    return redirect("/dashboard");
+  }
+
+  return { id, authority };
+};
+
+type LoaderData = {
+  id: string;
+  authority: Authority;
+};
+
+export const Details = () => {
+  const { id, authority } = useLoaderData() as LoaderData;
```

Comment:
```suggestion
  const { id, authority } = useLoaderData<typeof loaderData>();
```

This is how the [docs](https://api.reactrouter.com/v7/functions/react_router.useLoaderData.html) recommend typing it out

Reply:
I've been trying to make this work and after talking to Brian, it seems like we need to upgrade React Router to v7 for this to work 🤔

Reply:
Ask him if he minds a non-breaking change to v7:
https://reactrouter.com/upgrading/v6#upgrade-to-v7